docs/resources/samples/write-large-dataset.md

```diff
@@ -141,6 +143,119 @@ function getRandomString(length: number): string {
+
+### Sample code: Read part of a workbook
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook, startRow: number, batchSize: number) : string[][] {
+    // This sample only reads the first worksheet in the workbook.
+    const sheet = workbook.getWorksheets()[0];
+
+    // Get the boundaries of the range.
+    // Note that we're assuming usedRange is too big to read or write as a single Range.
+    const usedRange = sheet.getUsedRange();
+    const lastColumnIndex = usedRange.getLastColumn().getColumnIndex();
+    const lastRowindex = usedRange.getLastRow().getRowIndex();
+
+    // If we're starting past the last row, exit the script.
+    if (startRow > lastRowindex) {
+        return [[]];
+    }
+
+    // Get the next batch or the rest of the rows, whichever is smaller.
+    const rowCountToRead = Math.min(batchSize, (lastRowindex - startRow + 1));
+    const rangeToRead = sheet.getRangeByIndexes(startRow, 0, rowCountToRead, lastColumnIndex + 1);
+    return rangeToRead.getValues() as string[][];
+}
+
+```
+
+### Sample code: Write part of a workbook
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook, data: string[][], currentRow: number, batchSize: number): boolean {
+  // Get the first worksheet.
+  const sheet = workbook.getWorksheets()[0];
+
+  // Set the given data.
+  if (data && data.length > 0) {
+    sheet.getRangeByIndexes(currentRow, 0, data.length, data[0].length).setValues(data);
+  }
+
+  // If we wrote less data than the batch size, signal the end of the flow.
+  return batchSize > data.length;
+}
+```
+
+### Power Automate flow: Read and write data in a loop
+
+1. Sign into [Power Automate](https://flow.microsoft.com) and create a new **Instant cloud flow**.
+1. Choose **Manually trigger a flow** and select **Create**.
+1. Add a **New step** to track the current row being read and written. Make a new **Initialize variable** action with the following values.
+    * **Name**: currentRow
+    * **Type**: Integer
+    * **Value**: 0
+
+    :::image type="content" source="../../images/write-large-dataset-1.png" alt-text="The completed 'Initialize variable' step for the 'currentRow'.":::
+1. Add a **New step** to set the number of rows to be read in a single batch. Depending on the number of columns, this made need to be smaller to avoid the data transfer limits. Make a new **Initialize variable** action with the following values.
+    * **Name**: batchSize
+    * **Type**: Integer
+    * **Value**: 10000
+
+    :::image type="content" source="../../images/write-large-dataset-2.png" alt-text="The completed 'Initialize variable' step for the 'batchSize'.":::
+1. Add a **Do until** control. The flow will read chunks of the data until it has all been copied. You'll use the value of **-1** to indicate the end of the data has been reached. Give the control the following values.
+    * *First value*: *currentRow* (dynamic content)
```

Comment:
For consistency with the other steps, I think First value, Condition, and Second value should be bolded instead of italicized in this section. Like this: 
```suggestion
    * **First value**: *currentRow* (dynamic content)
```

Reply:
I debated about this. The other bolded words refer to a specific label in the UX, whereas the first term "First value" is never actually displayed. Do you think I should still make it bold?

Reply:
That's a good compromise. That would mean that in these walkthroughs bold text is actual labels, italics are dynamic content, and regular text is conceptual spaces or user input.

Reply:
I prefer the second option, I think it's clearer as a reader.